commons/src/main/java/org/dhis2/commons/dialogs/MediaDialog.kt

```diff
@@ -0,0 +1,341 @@
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import org.dhis2.commons.R
+import kotlin.random.Random
+
+const val SPACE_STRING = " "
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Preview
+@Composable
+private fun PreviewMediaDialog() {
+    MediaDialog(
+        title = "Tem local para lavar as mãos",
+        subTitle = "Pode ser uma torneira com água canalizada ou recepientes",
+        listOfAudioUrl = generateRandomURLs(),
+        listOfVideosUrl = generateRandomURLs(),
+        onMediaItemClicked = {},
+        onDismiss = {},
+    )
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Preview(backgroundColor = 0xFFFFFFFF)
+@Composable
+private fun PreviewVideoMediaItem() {
+    MediaDialogItem(
+        type = MediaType.VIDEO,
+        title = "Como lavar as mãos",
+        url = generateRandomURLs()[0],
+        duration = generateRandomDuration(),
+        dateOfLastUpdate = generateRandomDate()
+    ) {}
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Preview(backgroundColor = 0xFFFFFFFF)
+@Composable
+private fun PreviewAudioMediaItem() {
+    MediaDialogItem(
+        type = MediaType.AUDIO,
+        title = "Como lavar as mãos",
+        url = generateRandomURLs()[0],
+        duration = generateRandomDuration(),
+        dateOfLastUpdate = generateRandomDate()
+    ) {}
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun MediaDialog(
+    title: String,
+    subTitle: String,
+    listOfVideosUrl: List<String>, //Todo: create and use different object instead of String
```

Comment:
You could replace the list for each media type with a generic one.
For instance, you could create a generic data class that  contains generic listing and mediatype:

data classe MediaEntity<T>(
    data: List<T>,
    mediaType: MediaType
) 

or: 

data classe MediaEntity(
    data: List<String>,
    mediaType: MediaType
)